kpis/kpi_calculator.py

```diff
@@ -100,27 +100,40 @@ def get_thermal_discomfort(self, plot=False):
 
         '''
         
-        # Load temperature set points from test case data
         index=self.case.y_store['time']
-        LowerSetp = np.array(self.case.data_manager.get_data(index=index)
-                             ['LowerSetp'])
-        UpperSetp = np.array(self.case.data_manager.get_data(index=index)
-                             ['UpperSetp']) 
+        
         tdis_tot = 0
         tdis_dict = OrderedDict()
-        for signal in self.case.kpi_json['AirZoneTemperature']:
-            data = np.array(self.case.y_store[signal])
-            dT_lower = LowerSetp - data
-            dT_lower[dT_lower<0]=0
-            dT_upper = data - UpperSetp
-            dT_upper[dT_upper<0]=0
-            tdis_dict[signal[:-1]+'dTlower_y'] = \
-                trapz(dT_lower,self.case.y_store['time'])/3600.
-            tdis_dict[signal[:-1]+'dTupper_y'] = \
-                trapz(dT_upper,self.case.y_store['time'])/3600.
-            tdis_tot = tdis_tot + \
-                      tdis_dict[signal[:-1]+'dTlower_y'] + \
-                      tdis_dict[signal[:-1]+'dTupper_y']
+        
+        for source in self.case.kpi_json.keys():
+            if source.startswith(self.sources[0]):
+                # This is a potential source of thermal discomfort
+                zone_id = source.replace(self.sources[0]+'[','')[:-1]
```

Comment:
@JavierArroyoBastida Why change from referencing source by name to index of ``self.source``?  If by index, order of listing matters for future changes.

Reply:
@dhblum I myself was doubting on whether to do it with index or name. The benefit that I saw behind using the index was to have the source names in only one place so that changing the names wouldn't affect the rest of the code. But indeed, in this setting order of listing matters and may be less intuitive. I'll change it to refer source by name.

Reply:
I see what you're saying and agree there is a tradeoff.  Although, I think using the name makes the code more readable.  Also, we have the names separately listed in the Modelica IBPSA Library anyway.

Reply:
Addressed in https://github.com/ibpsa/project1-boptest/pull/178.